plugins/ltac2/tac2core.ml

```diff
@@ -675,7 +677,10 @@ let () =
   | (18, [|f|]) ->
     let f = Tac2ffi.to_float f in
     EConstr.mkFloat f
-  | (19, [|u;t;def;ty|]) ->
+  | (19, [|s|]) ->
+    let s = Tac2ffi.to_string s in
+    EConstr.mkString s
+  | (20, [|u;t;def;ty|]) ->
```

Comment:
That's where the problem is, to be explicit.

Reply:
(My local fix include a new Ltac2 `pstring` type, and a new  `Ltac2.Pstring` module defining functions on it.)

Reply:
OK, I pushed a fix, and I'm adapting overlays.

Reply:
I'm no Ltac2 expert but the solution sounds reasonable to me.
About the fact that it is possible to forge with plugins terms that would enable to violate the <= max_length axiom, this was already the case with primitive arrays IIUC so nothing really new.

Reply:
I see, OK. At least now it's not possible anymore with primitive strings!

Reply:
CC @SkySkimmer in case you want to have a look at the Ltac2 stuff.